link.html

```diff
@@ -252,26 +252,100 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
+              <label class="form-check-label text-dark" for="source-git">
+                Git Repository(eg. github)
+              </label>
+            </div>
+            <div class="form-check">
+              <input class="form-check-input" type="radio" name="source" id="source-googledrive" value="googledrive">
+              <label class="form-check-label text-dark" for="source-googledrive">
+                Google Drive
+              </label>
+            </div>
+            <div class="form-check">
+              <input class="form-check-input" type="radio" name="source" id="source-dropbox" value="dropbox">
+              <label class="form-check-label text-dark" for="source-dropbox">
+                Dropbox
+              </label>
+            </div>
+            <div class="form-check">
+              <input class="form-check-input" type="radio" name="source" id="source-generic-web" value="generic_web">
+              <label class="form-check-label text-dark" for="source-generic-web">
+                Web(Public URL to archive)
+              </label>
+            </div>
+          </div>
+        </div>
+        
+        
+        
+        <div class="form-group row source source-git" id="content-repo-group">
           <label for="content-repo" class="col-sm-2 col-form-label">Git Content Repository URL</label>
           <div class="col-sm-6">
             <input class="form-control" type="text" id="content-repo" placeholder="https://github.com/example/test"
-              oninput="displayLink()" pattern="((git|https?)://.+|git@.+:.+)">
+              oninput="displayLink()" required pattern="((git|https?)://.+|git@.+:.+)">
             <div class="invalid-feedback">
               Must be a valid git URL
             </div>
           </div>
           <div class="col-sm-4">
-            <div class="input-group" id="content-branch-group" hidden="true">
+            <div class="input-group" id="content-branch-group" >
               <div class="input-group-prepend">
                 <span class="input-group-text" id="content-branch-prepend-label">branch</span>
               </div>
-              <input name="content-branch" id="content-branch" type="text" class="form-control" value="master" aria-label="Branch Name" aria-describedby="content-branch-prepend-label">
+              <input name="content-branch" id="content-branch" type="text" placeholder="main" aria-label="Branch Name" aria-describedby="content-branch-prepend-label">
+              <small id="default-content-message" class="form-text text-muted">
+                 If left blank, the default branch of your repository is used. Also note, Github now names the default branch <code>main</code> instead of <code>master</code> on
+                 <a href="https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/">
+                 new GitHub repositories</a>
```

Comment:
```suggestion
                If left blank, the default branch of your repository is used. Also note, Github now names the default branch <code>main</code> instead of <code>master</code> on
                <a href="https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/">
                new GitHub repositories</a>
```